src/generator/AutoRest.Ruby/Templates/ServiceClientTemplate.cshtml

```diff
@@ -79,6 +79,67 @@ module @Settings.Namespace
     end
 
     @EmptyLine
+    #
+    # Makes a request and returns the body of the response.
+    # @@param method [Symbol] with any of the following values :get, :put, :post, :patch, :delete. 
+    # @@param path [String] the path, relative to {base_url}.
+    # @@param options [Hash{String=>String}] specifying any request options like :body.
+    # @@return [Hash{String=>String}] containing the body of the response.
+    # Example:
+    #
+    #  request_content = "{'location':'westus','tags':{'tag1':'val1','tag2':'val2'}}"
+    #  path = "/path"
+    #  options = {
+    #    body: request_content,
+    #    query_params: {'api-version' => '2016-02-01'}
+    #  }
+    #  result  = @@client.make_request(:put, path, options)
+    #    
+    def make_request(method, path, options = {})
+      result = make_request_async(method, path, options).value!
+      result.response.body.to_s.empty? ? nil : JSON.load(result.response.body)
+    end
+    @EmptyLine
+
+    #
+    # Makes a request asynchronously.
+    # @@param method [Symbol] with any of the following values :get, :put, :post, :patch, :delete.
+    # @@param path [String] the path, relative to {base_url}.
+    # @@param options [Hash{String=>String}] specifying any request options like :body.
+    # @@return [Concurrent::Promise] Promise object which holds the HTTP response.
+    #
+    def make_request_async(method, path, options = {})
```

Comment:
just a minor thought, do you think we should re-order the newly added three methods as first `make_request`, second `make_request_with_http_info` and third `make_request_async` as first two are internally calling third? 

Additionally make `make_request` to call `make_request_with_http_info` and then only return body?